### ISSUE TITLE
[Dreambooth] Training script mean loss calculation question

### ISSUE THREAD

**Issue by zetyquickly** [huggingface/diffusers#1273]
Hello,

I was playing with Dreambooth training and noticed a thing [here](https://github.com/huggingface/diffusers/blob/0025626cd9feca260e05f295f272af4f7b9ce44a/examples/dreambooth/train_dreambooth.py#L606) 

Why did the author use this way of calculation mean of the loss? Is there any particular technical need to do so comparing to [this](https://github.com/huggingface/diffusers/blob/0025626cd9feca260e05f295f272af4f7b9ce44a/examples/dreambooth/train_dreambooth.py#L609)

**Comment by patrickvonplaten** [comment]
cc @patil-suraj - could you take a look please? :-)

**Comment by zetyquickly** [comment]
One can tweak `prior_loss_weight` to change the weights in the sum of  the loss from instance samples and class samples

**Comment by patrickvonplaten** [comment]
cc @williamberman @pcuenca

**Comment by patil-suraj** [comment]
Hey @zetyquickly , I added it that way because this is how it was done in the original codebase, cf https://github.com/CompVis/stable-diffusion/blob/main/ldm/models/diffusion/ddpm.py#L307

But as @krahnikblis said 

```python
mse_goofy_way = F.mse_loss(test1.float(), test2.float(), reduction="none").mean([1, 2, 3]).mean()
mse_easy_way = F.mse_loss(test1.float(), test2.float(), reduction="mean")
```

both of these behave the same way, will open a PR to correct this and run some experiments to make sure.

**Comment by patil-suraj** [comment]
PR is here #1861
